functions/src/offers/index.ts

```diff
@@ -33,35 +34,37 @@ const queueStatusUpdateTriggers = (
 };
 
 const validateOffer = (value: IOffer): Promise<void> => {
-  return validateOrReject(Offer.factory(value)).then(() => {
-    return Promise.resolve();
-  });
+  return validateOrReject(Offer.factory(value))
+    .then(() => {
+      return Promise.resolve();
+    });
 };
 
 export const triggerEventsWhenOfferIsCreated = functions.firestore
   .document('offers/{offerId}')
   .onCreate((snapshot: DocumentSnapshot, context: EventContext) => {
-    return validateOffer(snapshot.data() as IOffer).catch(errors => {
-      console.error('Invalid Offer Found: ', errors);
-      return firestore()
-        .collection('offers')
-        .doc(context.params.offerId)
-        .delete();
-    });
+    return validateOffer(snapshot.data() as IOffer)
+      .catch(errors => {
+        console.error('Invalid Offer Found: ', errors);
+        return db
+          .collection('offers')
+          .doc(context.params.offerId)
+          .delete();
+      });
```

Comment:
just a formatting change to make the promises easier to read as a `catch` not as a `then`